### PR TITLE
Fallback to 24-hour time notation for locales that do not use 12-hour notation

### LINKED ISSUE #384: Spike on 12 hour time input in locales that use 24 hour format
# Feature Request

## Description
The translation team has noted that `en-US`, `es-US`, and `ar` are the only locales from our supported locale list that should include 12 hour display. Some guidance is needed on how we should handle the 12 hour time input when the locale is set to one that only uses 24 hour time formatting. Currently, we display 12-hour input, with the final input for a.m./p.m. being empty. Logging an issue so we can spike on this with the translation team / UX team to figure out a reasonable solution.

### PR DESCRIPTION
**Pull request by benbcai**
### Summary
Fall back to use the 24-hour time notation for locales that do not use the 12-hour notation.

### Additional Details
- Fixes #384 
- The removal of the am/am strings in the translation files is what is expected but still waiting for the translations team for confirmation.

### Deployment Links:

* [time-input](https://terra-framework-deploye-pr-898.herokuapp.com/components/terra-time-input/time-input/time-input)

### CODE REVIEW

File: packages/terra-time-input/src/TimeInput.jsx [hunk 1]
```diff
@@ -137,7 +137,7 @@ class TimeInput extends React.Component {
     let hour = TimeUtil.splitHour(value);
     let meridiem;
 
-    if (props.variant === TimeUtil.FORMAT_12_HOUR) {
```
Comment (emilyrohrbough) [inline packages/terra-time-input/src/TimeInput.jsx]: We could reduce this call if we assigned the value:

```diff
+ const variant = TimeUtil.getVariantFromLocale(this.props)

 if (
   this.props.value === prevProps.value
+  && TimeUtil.getVariantFromLocale(this.props) === TimeUtil.getVariantFromLocale(prevProps)
-  && variant=== TimeUtil.getVariantFromLocale(prevProps)
 ) {
  return;
 }


   let hour = TimeUtil.splitHour(this.props.value);
   let { meridiem } = this.state;

- if (TimeUtil.getVariantFromLocale(this.props) === TimeUtil.FORMAT_12_HOUR) {
+ if (variant === TimeUtil.FORMAT_12_HOUR) {
```

Comment (benbcai) [reply packages/terra-time-input/src/TimeInput.jsx]: I think this comment was intended for the changes made in `componentDidUpdate`. Updated.

File: packages/terra-time-input/src/TimeUtil.js [hunk 2]
```diff
@@ -253,6 +253,21 @@ class TimeUtil {
 
     return /^(0[0-9]|1[0-9]|2[0-3]):[0-5][0-9]$/.test(value);
   }
+
+  /**
+   * Determines if the locale can use the 12-hour time notation. Default to use the 24-hour variant if the locale does not use it.
+   * @param {Object} props - The component props.
+   * @return {String} FORMAT_24_HOUR if the locale does not use the 12-hour notation. Otherwise, the current variant prop.
+   */
+  static getVariantFromLocale(props) {
```
Comment (emilyrohrbough) [inline packages/terra-time-input/src/TimeUtil.js]: can we add a jest test for this new utility method?

Comment (benbcai) [reply packages/terra-time-input/src/TimeUtil.js]: Added test: https://github.com/cerner/terra-framework/pull/898/commits/fee009ceaf38a7b2ad8fb36ccf3be154c9b148da

**Comment by mmalaker** [comment]
Functional verification completed. The 12 hour time input examples display and function as expected in each locale.

**Pull request merged**